Flaky-test: LookupPropertiesTest.testConcurrentLookupProperties

### Search before reporting

- [x] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/18398680856/job/52423315315?pr=24836#step:11:1638

### Exception stacktrace

```text
Error:  Tests run: 5, Failures: 1, Errors: 0, Skipped: 4, Time elapsed: 28.47 s <<< FAILURE! -- in org.apache.pulsar.client.api.LookupPropertiesTest
  Error:  org.apache.pulsar.client.api.LookupPropertiesTest.testConcurrentLookupProperties -- Time elapsed: 2.096 s <<< FAILURE!
  java.lang.AssertionError: Lists differ at element [0]: key-3 != key-0 expected [key-3] but found [key-0]
  	at org.testng.Assert.fail(Assert.java:110)
  	at org.testng.Assert.failNotEquals(Assert.java:1577)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:149)
  	at org.testng.Assert.assertEquals(Assert.java:1643)
  	at org.testng.Assert.assertEquals(Assert.java:1605)
  	at org.apache.pulsar.client.api.LookupPropertiesTest.testConcurrentLookupProperties(LookupPropertiesTest.java:115)
  	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
  	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  	at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/client/api/LookupPropertiesTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/client/api/LookupPropertiesTest.java
@@ -18,6 +18,7 @@
  */
 package org.apache.pulsar.client.api;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import java.net.InetSocketAddress;
 import java.util.ArrayList;
 import java.util.Collections;
@@ -112,7 +113,7 @@ public class LookupPropertiesTest extends MultiBrokerBaseTest {
             client.getConfiguration().setLookupProperties(Collections.emptyMap());
         }
         FutureUtil.waitForAll(futures).get();
-        Assert.assertEquals(clientIdList, BrokerIdAwareLoadManager.CLIENT_ID_LIST);
+        assertThat(BrokerIdAwareLoadManager.CLIENT_ID_LIST).containsExactlyInAnyOrderElementsOf(clientIdList);
     }
 
     public static class BrokerIdAwareLoadManager extends ExtensibleLoadManagerImpl {

